src/ifs/wl_seat/event_handling.rs

```diff
@@ -234,6 +241,34 @@ impl WlSeatGlobal {
         }
     }
 
+    fn touch_event(self: &Rc<Self>, id: i32, time_usec: u64, event: TouchEvent) {
+        let pos_transformed = |pos: TouchPosition| {
+            let output_pos = self.output.get().global.pos.get();
+            let x = Fixed::from_f64(pos.x_transformed.to_f64() * f64::from(output_pos.width()));
+            let y = Fixed::from_f64(pos.y_transformed.to_f64() * f64::from(output_pos.height()));
+            (x, y)
+        };
```

Comment:
`self.output` is the output that currently contains the pointer. If there are multiple outputs, then this code is not going to work. For any touch input device, touch input should always map to a specific output. There will need to be a new configuration option to map input devices to outputs.

Reply:
I added a config option `mapped-output` and a fallback to the built-in output, if any.